### PR TITLE
Add minidump collection via breakpad

### PR DESCRIPTION
**Pull request by driazati**
This adds [breakpad](https://github.com/google/breakpad) which comes with out-of-the-box utilities to register a signal handler that writes out a minidump on an unhandled exception. Right now this is gated behind a flag in `torch.utils`, but in the future it could be on by default. Sizewise this adds aboute 500k to `libtorch_cpu.so` (187275968 B to 187810016 B). 

```bash
$ cat <<EOF > test.py
import torch

torch.utils.enable_minidump_collection()

# temporary util that just segfaults
torch._C._crash()
EOF

$ python test.py
Wrote minidump to /tmp/pytorch_crashes/6a829041-50e9-4247-ea992f99-a74cf47a.dmp
fish: “python test.py” terminated by signal SIGSEGV (Address boundary error)
$ minidump-2-core /tmp/pytorch_crashes/6a829041-50e9-4247-ea992f99-a74cf47a.dmp -o core.dmp
$ gdb python core.dmp
... commence debugging ...
```

Right now all exceptions that get passed up to Python don't trigger the signal handler (which by default only 
handles [these](https://github.com/google/breakpad/blob/main/src/client/linux/handler/exception_handler.cc#L115)). It would be possible for PyTorch exceptions to explicitly write a minidump when passed up to Python (maybe only when the exception is unhandled or something).

TODO:
* Figure out build weirdness (breakpad uses autotools and also requires a header to be downloaded into its source tree before it will build correctly)

### CODE REVIEW

File: torch/CMakeLists.txt [hunk 1]
```diff
@@ -65,6 +66,7 @@ set(TORCH_PYTHON_INCLUDE_DIRECTORIES
     ${CMAKE_BINARY_DIR}/caffe2/aten/src
     ${CMAKE_BINARY_DIR}/third_party
     ${CMAKE_BINARY_DIR}/third_party/onnx
+    ${CMAKE_BINARY_DIR}/third_party/breakpad/src
```
Comment (malfet) [inline torch/CMakeLists.txt:69]: This is probably not needed (TORCH_PYTHON_INCLUDE_DIRECTORIES is public dependencies, but breakpad should be private)

File: torch/csrc/utils/crash_handler.cpp [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+               nullptr,
+               true,
+               -1);
+         }
+       })
+      .def("_get_minidump_directory", []() { return minidump_directory; })
+      .def("_crash", []() {
+        // TODO: Testing only, remove before landing
+        volatile int* bad = nullptr;
+        return *bad;
```
Comment (malfet) [inline torch/csrc/utils/crash_handler.cpp:46]: Can this perhaps be unified with  "_crash_if_aten_asan" (implemented in torch/csrc/Modules.cpp)

Comment (driazati) [reply torch/csrc/utils/crash_handler.cpp]: Removed it in lieu of an actual test

**Comment by driazati** [comment]
Closing in favor of #55647

**Pull request closed without merging**